### ISSUE TITLE
B007 Loop control variable '_' not used within loop body

### ISSUE THREAD

**Issue by montanarograziano** [astral-sh/ruff#13921]
Hi, I don't know if this is an intended behaviour or not, but I'm having troubles understanding why a snippet like the following is highlighted as [B007](https://docs.astral.sh/ruff/rules/unused-loop-control-variable/)
```py
for _ in range(10)
    print("hello")
```
From the doc, any variable prefixed with `_` should be safe. Is there anything I'm doing wrong?
I know the example is (way too) trivial, but a more realistic use-case would be cycling through rows in a Pandas DataFrame with `iterrows()` but without using the index.
Thanks for the help!

**Comment by autinerd** [comment]
Do you have some context where it gets flagged?

When testing here https://play.ruff.rs/56729c12-30db-43e0-955c-2e82ac2c4ece, it shows no violation.

**Comment by montanarograziano** [comment]
I get this problem only when using Ruff under pre-commit, which is configured to use my local ruff version (0.6.4)
Here's a screenshot:
<img width="816" alt="image" src="https://github.com/user-attachments/assets/7ee33840-d6b7-4300-84c1-c9ab9f72ebec">

**Comment by montanarograziano** [comment]
Mystery solved! Apparently, there was a typo in my team's config `dummy-variable-rgx` inside pyproject.toml 😅
Sorry about the false positive and thank you for the help.
